packages/oauth2/src/client-attestation-pop.ts

```diff
@@ -131,6 +136,13 @@ export async function createClientAttestationPopJwt(
     );
   }
 
+  const sub = clientAttestation.payload.sub as string | undefined;
+  if (!sub) {
```

Comment:
```suggestion
  const sub = clientAttestation.payload.sub ;
  if (!sub || typeof sub !== 'string') {
```
I think this check in its original form only prevented `sub` from being `undefined`, but didn't check if it was actually a `string` or not, this way `sub` will certainly be a string.